docs/evm/cadence/batched-evm-transactions.md

```diff
@@ -0,0 +1,641 @@
+        self.wflowAddress = EVM.addressFromString(wflowAddressHex)
+
+        /* Assign the ERC721 EVM Address */
+        //
+        // Deserialize the provided ERC721 hex string to an EVM address
+        self.erc721Address = EVM.addressFromString(maybeMintERC721AddressHex)
+    }
+
+    pre {
+        self.coa.balance().inFLOW() >= self.mintCost:
+            "CadenceOwnedAccount holds insufficient FLOW balance to mint - "
+            .concat("Ensure COA has at least ".concat(self.mintCost.toString()).concat(" FLOW"))
+    }
+
+    execute {
+        /* Wrap FLOW in EVM as WFLOW */
+        //
+        // Encode calldata & set value
+        let depositCalldata = EVM.encodeABIWithSignature("deposit()", [])
+        let value = EVM.Balance(attoflow: 0)
+        value.setFLOW(flow: self.mintCost)
+        // Call the WFLOW contract, wrapping the sent FLOW
+        let wrapResult = self.coa.call(
+            to: self.wflowAddress,
+            data: depositCalldata,
+            gasLimit: 15_000_000,
+            value: value
+        )
+        assert(
+            wrapResult.status == EVM.Status.successful,
+            message: "Wrapping FLOW as WFLOW failed: ".concat(wrapResult.errorMessage)
+        )
+
+        /* Approve the ERC721 address for the mint amount */
+        //
+        // Encode calldata approve(address,uint) calldata, providing the ERC721 address & mint amount
+        let approveCalldata = EVM.encodeABIWithSignature(
+                "approve(address,uint256)",
+                [self.erc721Address, UInt256(1_000_000_000_000_000_000)]
+            )
+        // Call the WFLOW contract, approving the ERC721 address to move the mint amount
+        let approveResult = self.coa.call(
+            to: self.wflowAddress,
+            data: approveCalldata,
+            gasLimit: 15_000_000,
+            value: EVM.Balance(attoflow: 0)
+        )
+        assert(
+            approveResult.status == EVM.Status.successful,
+            message: "Approving ERC721 address on WFLOW contract failed: ".concat(approveResult.errorMessage)
+        )
+
+        /* Attempt to mint ERC721 */
+        //
+        // Encode the mint() calldata
+        let mintCalldata = EVM.encodeABIWithSignature("mint()", [])
+        // Call the ERC721 contract, attempting to mint
+        let mintResult = self.coa.call(
+            to: self.erc721Address,
+            data: mintCalldata,
+            gasLimit: 15_000_000,
+            value: EVM.Balance(attoflow: 0)
+        )
+        // If mint fails, all other actions in this transaction are reverted
+        assert(
+            mintResult.status == EVM.Status.successful,
+            message: "Minting ERC721 token failed: ".concat(mintResult.errorMessage)
+        )
+    }
+}
+
+```
+</details>
+
+You can run the transaction at the following link using the community-developed Flow Runner tool:
+[`wrap_and_mint.cdc`](https://run.dnz.dev/snippet/4fe0d43fdaf968b9).
```

Comment:
I've got lots of errors in Flow runner.  Is this expected?  Can we fix it?

Reply:
And I was missing the window for arguments.  Came back after refreshing.  Anything we can do about that?  Is this common?

Reply:
Worth reporting to @bluesign (maintainer for the tool). I think it may be a matter of the language server having a chance to pull dependencies from Testnet/Mainnet